### ISSUE TITLE
Copy geocache link from geocache page

### ISSUE THREAD

**Issue by king-ton** [cgeo/cgeo#9339]
**Describe the bug:**
Copy a link for geocache if a geocache is open.
Since some update it is not possible any more to copy a link to the open geocache. If you select the cache on the map it is no problem to copy a link for this cache. But there open different dialogs.
It is possible to include this feature again to get a link to this geocache?

**To Reproduce:**
1. Open the geocache
2. Select 3 dots in right upper corner
3. Redirect >

**Actual behavior/state after performing these steps:**
Copy to clipboard is missing

**Expected behavior/state after performing these steps:**
There should be an entry with copy link to clipboard

**Version of c:geo used:**
2020.10.29

**System information:**
OnePlus 5
Android 10

**Comment by fm-sys** [comment]
You can try following app: https://play.google.com/store/apps/details?id=com.robglez.copyshare

It is free and without ads. As the "copy link" feature is also missing for most other apps inside the share menu, it is not really a c:goe bug imho.

**Comment by Lineflyer** [comment]
But interesting enough:
The share menu in the popup and the share menu from cache details lead to different sharing menus.

I personally prefer the one from the popup (as it is more complete and also contains clipboard) over the other one in cache details (which IMHO looks a bit broken on my device).

**Comment by king-ton** [comment]
> But interesting enough:
> The share menu in the popup and the share menu from cache details lead to different sharing menus.
> 
> I personally prefer the one from the popup (as it is more complete and also contains clipboard) over the other one in cache details (which IMHO looks a bit broken on my device).

Same for me.

**Comment by fm-sys** [comment]
The share menu at the live map is provided by Android while the gui for sharing at the cache details screen is created our self. But interesting for me: At the details page 47 apps are displayed for sharing, but at the live map only 42!? 

![Screenshot_20201102_160058_com huawei android internal app](https://user-images.githubusercontent.com/64581222/97885522-c0c50700-1d27-11eb-8080-f3bad6e0dac9.jpg) ![Screenshot_20201102_162233_cgeo geocaching](https://user-images.githubusercontent.com/64581222/97885525-c15d9d80-1d27-11eb-821c-9780971fb705.jpg)

pro android system internal:
-  is the appreciated one by Android 
-  the developers of the other apps must make sure that they appear in this menu, not we
- design is more known by other apps
- c:geo's seem to be a bit broken (see screenshot)

contra:
- not as many apps are displayed at once so more scrolling is needed (at least on my device)
- not all apps seem to appear in this menu

**Comment by Lineflyer** [comment]
I vote for the system standard as users know this design. 

The question would be, why we implemented it in a different way (and which apps are missing)... AFAIR we used to use the "normal" share menu times ago.

**Comment by cayacdev** [comment]
I would vote also for the system standard.

I investigated some time and found out that `androidx.appcompat.widget.ShareActionProvider` is used in the cache detail page. For some reason this provider is null on the map view. If it is null the method `cgeo.geocaching.models.Geocache#shareCache` is called. Therefore I removed `androidx.appcompat.widget.ShareActionProvider` and all related methods and call `cgeo.geocaching.models.Geocache#shareCache` every time. 

I tested it shortly, it seems that it works now and the system internal cache menu is used in both views.

I can created a PR with this changes. What other functions should I test before?

<img src="https://user-images.githubusercontent.com/6408145/99112418-f0102980-25ed-11eb-844e-2bb48bae1bfb.png" width="300"><img src="https://user-images.githubusercontent.com/6408145/99112471-01f1cc80-25ee-11eb-8f58-e5fa771d70d1.png" width="300">
